src/backend/marsha/core/utils/xmpp_utils.py

```diff
@@ -0,0 +1,177 @@
+            frm=settings.XMPP_PRIVATE_SERVER_JID,
+            typ="set",
+            queryNS=xmpp.NS_MUC_OWNER,
+            payload=[
+                xmpp.DataForm(
+                    typ="submit",
+                    data=[
+                        xmpp.DataField(
+                            typ="hidden", name="FORM_TYPE", value=xmpp.NS_MUC_ROOMCONFIG
+                        ),
+                        # Make Room Persistent?
+                        xmpp.DataField(
+                            typ="boolean", name="muc#roomconfig_persistentroom", value=1
+                        ),
+                        # Make room publicly searchable?
+                        xmpp.DataField(
+                            typ="boolean", name="muc#roomconfig_publicroom", value=0
+                        ),
+                        # Nobody can send private message
+                        xmpp.DataField(
+                            typ="list-single",
+                            name="muc#roomconfig_allowpm",
+                            value="none",
+                        ),
+                        # Nobody can send private message
+                        xmpp.DataField(
+                            typ="boolean", name="muc#roomconfig_allowinvites", value=0
+                        ),
+                        # Nobody can change the subject
+                        xmpp.DataField(
+                            typ="boolean", name="muc#roomconfig_changesubject", value=0
+                        ),
+                    ],
+                )
+            ],
+        )
+    )
+
+
+def close_room(room_name):
+    """Close a room to anonymous user.
+
+    members only documentation:
+    https://xmpp.org/extensions/xep-0045.html#enter-members
+
+    Parameters
+    ----------
+    room_name: string
+        The name of the room you want to destroy.
+    """
+    client = _connect()
+
+    client.send(
+        xmpp.Presence(
+            to=f"{room_name}@{settings.XMPP_CONFERENCE_URL}/admin",
+            payload=[xmpp.Node(tag="x", attrs={"xmlns": xmpp.NS_MUC})],
+        )
+    )
+
+    client.send(
+        xmpp.Iq(
+            to=f"{room_name}@{settings.XMPP_CONFERENCE_URL}",
+            frm=settings.XMPP_PRIVATE_SERVER_JID,
+            typ="set",
+            queryNS=xmpp.NS_MUC_OWNER,
+            payload=[
+                xmpp.DataForm(
+                    typ="submit",
+                    data=[
+                        xmpp.DataField(
+                            typ="hidden", name="FORM_TYPE", value=xmpp.NS_MUC_ROOMCONFIG
+                        ),
+                        xmpp.DataField(
+                            typ="boolean", name="muc#roomconfig_membersonly", value=1
+                        ),
+                    ],
+                )
+            ],
+        )
+    )
+
+
+def _generate_expired_date():
+    """Generate a datetime object 1 day in the future."""
+    return timezone.now() + timedelta(days=1)
```

Comment:
This function exists only for testing purpose ?

Why not adding a new parameter (`expires_at`)  parameter to the `generate_jwt` function instead ?
It would be more flexible and easier to test IMO.

Reply:
Yes it's for testing purpose. And yes your idea is better :+1: